contracts/signature_mint/ERC1155/SignatureMint1155.sol

```diff
@@ -0,0 +1,467 @@
+            tokenIdToMint = _req.tokenId;
+        }
+
+        _mintTo(receiver, _req.uri, tokenIdToMint, _req.quantity);
+
+        collectPrice(_req, tokenIdToMint);
+
+        emit MintWithSignature(signer, receiver, tokenIdToMint, _req);
+    }
+
+    /// @dev See EIP 2981
+    function royaltyInfo(uint256, uint256 salePrice)
+        external
+        view
+        virtual
+        override
+        returns (address receiver, uint256 royaltyAmount)
+    {
+        receiver = controlCenter.getRoyaltyTreasury(address(this));
+        royaltyAmount = (salePrice * royaltyBps) / MAX_BPS;
+    }
+
+    //      =====   Setter functions  =====
+
+    /// @dev Lets a module admin set the recipient of all primary sales for a given token ID.
+    function setSaleRecipient(uint256 _tokenId, address _saleRecipient) external onlyModuleAdmin {
+        saleRecipient[_tokenId] = _saleRecipient;
+        emit NewSaleRecipient(_saleRecipient, _tokenId, false);
+    }
+
+    /// @dev Lets a module admin set the default recipient of all primary sales.
+    function setDefaultSaleRecipient(address _saleRecipient) external onlyModuleAdmin {
+        defaultSaleRecipient = _saleRecipient;
+        emit NewDefaultSaleRecipient(_saleRecipient);
+    }
+
+    /// @dev Lets a module admin update the royalties paid on secondary token sales.
+    function setRoyaltyBps(uint256 _royaltyBps) public onlyModuleAdmin {
+        require(_royaltyBps <= MAX_BPS, "bps <= 10000.");
+
+        royaltyBps = uint64(_royaltyBps);
+
+        emit RoyaltyUpdated(_royaltyBps);
+    }
+
+    /// @dev Lets a module admin update the fees on primary sales.
+    function setFeeBps(uint256 _feeBps) public onlyModuleAdmin {
+        require(_feeBps <= MAX_BPS, "bps <= 10000.");
+
+        feeBps = uint120(_feeBps);
+
+        emit PrimarySalesFeeUpdates(_feeBps);
+    }
+
+    /// @dev Lets a module admin restrict token transfers.
+    function setRestrictedTransfer(bool _restrictedTransfer) external onlyModuleAdmin {
+        transfersRestricted = _restrictedTransfer;
+
+        emit TransfersRestricted(_restrictedTransfer);
+    }
+
+    /// @dev Lets a module admin set a new owner for the contract. The new owner must be a module admin.
+    function setOwner(address _newOwner) external onlyModuleAdmin {
+        require(hasRole(DEFAULT_ADMIN_ROLE, _newOwner), "new owner not module admin.");
+        address _prevOwner = _owner;
+        _owner = _newOwner;
+
+        emit NewOwner(_prevOwner, _newOwner);
+    }
+
+    /// @dev Lets a module admin set the URI for contract-level metadata.
+    function setContractURI(string calldata _uri) external onlyModuleAdmin {
+        contractURI = _uri;
+    }
+
+    ///     =====   Internal functions  =====
+
+    /// @dev Mints an NFT to `to`
+    function _mintTo(address _to, string calldata _uri, uint256 _tokenId, uint256 _amount) internal {
+        
+        if(bytes(_tokenURI[_tokenId]).length == 0) {
+            _tokenURI[_tokenId] = _uri;
```

Comment:
we probably want to require the length of `_uri` > 0.
so you can't mint new token with empty uri.